modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationDependingOnDomainDeploymentTestCase.java

```diff
@@ -216,4 +224,70 @@ public void compatibleDomainNameIsSetAfterDeployment() throws Exception {
     ApplicationDescriptor applicationDescriptor = application.getDescriptor();
     assertThat(applicationDescriptor.getDomainName(), is(emptyDomain101FileBuilder.getId()));
   }
+
+  @Test
+  public void stoppedApplicationsAreNotStartedWhenDomainIsRedeployed() throws Exception {
+    DeploymentListener mockDeploymentListener = spy(new DeploymentStatusTracker());
```

Comment:
I think these two tests have duplicate code. Maybe they can be refactored taking into account this.